qutebrowser/mainwindow/tabwidget.py

```diff
@@ -102,11 +102,12 @@ def __init__(self, win_id, parent=None):
     def __repr__(self):
         return utils.get_repr(self, count=self.count())
 
-    @config.change_filter('tabs', 'auto-hide')
+    @config.change_filter('tabs', 'hide-auto')
     def autohide(self):
         """Auto-hide the tabbar if needed."""
```

Comment:
This won't get called when `hide-always` is updated because of the `@config.change_filter`, so the UI isn't updated automatically when the user uses `:set tabs hide-always true`.

Unfortunately `@config.change_filter` doesn't support multiple options, so I suggest writing a helper option:

```python
def on_hide_config_changed(self, section, option):
    """Hide tabbar if needed when the config is changed."""
    if section == 'tabs' and option in ('hide-auto', 'hide-always'):
        self.autohide()
```

Then you can remove the `@config.change_filter(...)` decorator here, and adjust the `config_obj.changed.connect(...)` call above.